### PR TITLE
removed  && __builtin_constant_p(V) in digitalWriteFast

### PR DESCRIPTION
**Pull request by ArminJo**
Removed  && __builtin_constant_p(V) in digitalWriteFast, since it prevents the usage of  BIT_WRITE(value, bit, bitvalue) (bitvalue ? BIT_SET(value, bit) : BIT_CLEAR(value, bit) if V is not constant.
But BIT_WRITE does not need a constant bitvalue (or V) to work. With a constant value it can be better optimized, but it works also without bitvalue (or V) being constant.

### CODE REVIEW

**Comment by ArminJo** [comment]
The comment for the second commit
*replaced `BIT_CLEAR`...*
was not correct. Correct is: 
*replaced `BIT_WRITE` ...*

**Comment by awatterott** [comment]
I have to check. As far as I remember the check was needed and maybe it works now on newer GCC versions.

**Comment by ArminJo** [comment]
as long as we're at it... ATTiny88 additions.
Thanks. 

Regarding the first change, the current arduino GCC does well optimizing, I looked carefuly at the assembler output.
(And this was the reason, why I stumbeld over it)

File: digitalWriteFast/digitalWriteFast.h [hunk 1]
```diff
@@ -369,11 +369,17 @@
...
+#  if defined(ARDUINO_AVR_DIGISPARKPRO)
+#define __digitalPinToPortReg(P) (((P) <= 7) ? &PORTD : ((P) <= 14 ? &PORTB : ((P) <= 18 ? &PORTA : &PORTC)))
+#define __digitalPinToDDRReg(P)  (((P) <= 7) ? &DDRD : ((P) <= 14 ? &DDRB : ((P) <= 18 ? &DDRA : &DDRC)))
+#define __digitalPinToPINReg(P)  (((P) <= 7) ? &PIND : ((P) <= 14 ? &PINB : ((P) <= 18 ? &PINA : &PINC)))
 #define __digitalPinToBit(P) ( ((P) <= 7) ? (P) : (((P) <= 13) ? ((P) - 8) : (((P) == 14) ? 7 : (((P) <= 16) ? ((P) - 14) : (((P) <= 18) ? ((P) - 17) : (((P) == 25) ? 7 : ((P) - 19)))))) )
-
+#  else
+#define __digitalPinToPortReg(P) (((P) <= 7) ?(&PORTD : (((P) <= 15) ? &PORTB : (((P) <= 22) ? &PORTC : (((P) <= 26) ?(&PORTA : &PORTC))))
+#define __digitalPinToPortReg(P) (((P) <= 7) ?(&DDRD : (((P) <= 15) ? &DDRB : (((P) <= 22) ? &DDRC : (((P) <= 26) ?(&DDRA : &DDRC))))
+#define __digitalPinToPortReg(P) (((P) <= 7) ?(&PIND : (((P) <= 15) ? &PINB : (((P) <= 22) ? &PINC : (((P) <= 26) ?(&PINA : &PINC))))
```
[the remark this answers is not in the document]
Comment (ArminJo) [reply digitalWriteFast/digitalWriteFast.h]: I corrected it and solved my braces bug.
Problem was, I have no board at hand.
I swear, next time I will test it better and maybe issue a new pull request for such additions.

**Comment by awatterott** [comment]
Sorry for the late reply. I have made tests today and everything is working.
Thank you for the improvements.

**Pull request merged**